ledger/src/blockstore.rs

```diff
@@ -3604,6 +3604,68 @@ impl Blockstore {
         self.get_slot_entries_in_block(slot, vec![(start_index, end_index)], slot_meta)
     }
 
+    /// Returns true if the last `DATA_SHREDS_PER_FEC_BLOCK` data shreds of a
+    /// slot have the same merkle root, indicating they are a part of the same
+    /// FEC set.
+    /// Will fail if:
+    ///     - Slot meta is missing
+    ///     - LAST_SHRED_IN_SLOT flag has not been received
+    ///     - There are missing shreds in the last fec set
+    ///     - The block contains legacy shreds
+    pub fn is_last_fec_set_full(&self, slot: Slot) -> Result<bool> {
+        // We need to check if the last FEC set index contains at least `DATA_SHREDS_PER_FEC_BLOCK` data shreds.
+        // We compare the merkle roots of the last `DATA_SHREDS_PER_FEC_BLOCK` shreds in this block.
+        // Since the merkle root contains the fec_set_index, if all of them match, we know that the last fec set has
+        // at least `DATA_SHREDS_PER_FEC_BLOCK` shreds.
+        let slot_meta = self
+            .meta_cf
+            .get(slot)?
+            .ok_or(BlockstoreError::SlotUnavailable)?;
+        let last_shred_index = slot_meta
+            .last_index
+            .ok_or(BlockstoreError::UnknownLastIndex(slot))?;
+
+        let expected_num_shreds = u64::try_from(DATA_SHREDS_PER_FEC_BLOCK)
+            .expect("DATA_SHREDS_PER_FEC_BLOCK should fit in u64");
+        let minimum_index = expected_num_shreds
+            .checked_sub(1)
+            .expect("DATA_SHREDS_PER_FEC_BLOCK should be non-zero");
```

Comment:
`DATA_SHREDS_PER_FEC_BLOCK` is a const.
You can just `const_assert` these instead of runtime checks.